src/providers/live/live-provider.ts

```diff
@@ -70,22 +68,47 @@ export class LiveProvider extends Provider {
     const id3TagCues = payload.cues.filter((cue: any) => cue.value && cue.value.key === 'TEXT');
     if (id3TagCues.length) {
       try {
-        const id3Data = JSON.parse(id3TagCues[id3TagCues.length - 1].value.data);
+        const lastCue = id3TagCues[id3TagCues.length - 1];
+        const id3Data = JSON.parse(lastCue.value.data);
+        const {timestamp, clipId, setId} = id3Data;
 
-        const id3Timestamp = Math.ceil(id3Data.timestamp / 1000);
+        const id3Timestamp = Math.ceil(timestamp / 1000);
         if (id3Timestamp) {
           this._id3Timestamp = id3Timestamp;
         }
 
-        if (!this._types.has(KalturaCuePointType.HOTSPOT) || !id3Data.clipId || !id3Data.setId) return;
-
-        const [partType, originalEntryId] = id3Data.clipId.split('-');
-        if (!this._simuliveClipIds.has(originalEntryId)) {
-          const cueOffset = this._getSimuliveCueOffset(id3Data.timestamp, id3Data.setId, id3Data.clipId, id3TagCues[id3TagCues.length - 1].startTime);
-          if (cueOffset === null) return;
+        if (!id3Data.clipId || !id3Data.setId) return;
+
+        const [partType, originalEntryId, clipStartTimestamp] = id3Data.clipId.split('-');
+        const cuepointOffset = this._getSimuliveCuepointOffset(
+          id3Data.timestamp,
+          id3Data.setId,
+          id3Data.clipId,
+          id3TagCues[id3TagCues.length - 1].startTime
+        );
+        if (cuepointOffset === null) return;
+
+        const textTracks = [...(this._player.getVideoElement().textTracks as any)];
+        const cuepointsTrack = textTracks.find(t => t.label === 'CuePoints');
+        const activeCues = cuepointsTrack?.activeCues;
+
+        if (activeCues) {
+          for (const cue of activeCues) {
+            if (
+              // @ts-ignore
+              cue.startTime < this._player.getStartTimeOfDvrWindow() + cuepointOffset &&
```

Comment:
maybe extract the calculation of `this._player.getStartTimeOfDvrWindow() + cuepointOffset` before the if condition? since it's repeated 3 times.